### ISSUE TITLE
Not able to run diff on two dataframe in databricks 

### ISSUE THREAD

**Issue by shri-0509** [G-Research/spark-extension#246]
here is the code

from gresearch.spark.diff import *
left = spark.createDataFrame([(1, "one"), (2, "two"), (3, "three")], ["id", "value"])
right = spark.createDataFrame([(1, "one"), (2, "Two"), (4, "four")], ["id", "value"])
print(spark.version)
left.diff(right).show()

Error: [[ATTRIBUTE_NOT_SUPPORTED](https://docs.microsoft.com/azure/databricks/error-messages/error-classes#attribute_not_supported)] Attribute `diff` is not supported.

Spark version-3.5.0
have added maven library: uk.co.gresearch.spark:spark-extension_2.12:2.12.0-3.5

**Comment by EnricoMi** [comment]
Looks like Databricks does not like how we extend `DataFrame` with `.diff`.

You can diff as follows:

```python
diff(left, right)
```

**Comment by EnricoMi** [comment]
Maybe `spark.createDataFrame` does not reurn a `pyspark.sql.dataframe.DataFrame` but some Databricks `DataFrame`.

Could you please execute the following on your side and share the output?

    print(type(left))

**Comment by shri-0509** [comment]
yes it will result in different dataframe <class 'pyspark.sql.connect.dataframe.DataFrame'>.

**Comment by EnricoMi** [comment]
I managed to reproduce the issue with a local Spark Connect server. Looks like the diffing does not work with Spark Connect. Will investigate a fix.

**Comment by shri-0509** [comment]
sure thanq. right now i am doing left_anti join to get added, deleted and inner join to get modified and unchanged. Thinking to use this library to do the same
